roles/ec2_instance_create/README.md

```diff
@@ -0,0 +1,132 @@
+# ec2_instance_create
+
+A role to create an EC2 instance in AWS.
+
+Users can specify various parameters for instance configuration, including instance type, AMI ID, key pair, tags, and VPC/subnet configuration.
+
+This role also supports the creation of optional networking resources, such as an external security group and an Elastic IP (EIP). You can choose to wait for the EC2 instance to finish booting before continuing.
+
+## Role Variables
+
+The following variables can be set in the role to customize EC2 instance creation and networking configurations:
+
+### EC2 Instance Configuration
+
+* **ec2_instance_create_aws_region**: (Required)
+  The AWS region in which to create the EC2 instance.
+
+* **ec2_instance_create_instance_name**: (Required)
+  The name of the EC2 instance to be created.
+
+* **ec2_instance_create_instance_type**: (Required)
+  The instance type for the EC2 instance (e.g., `t2.micro`, `m5.large`).
+
+* **ec2_instance_create_ami_id**: (Required)
+  The AMI ID for the EC2 instance.
+
+* **ec2_instance_create_key_name**: (Required)
+  The name of the key pair to use for SSH access to the EC2 instance.
+
+* **ec2_instance_create_vpc_subnet_id**: (Required)
+  The ID of the VPC subnet in which the instance will be launched.
+
+  * **ec2_instance_create_vpc_id**: (Optional)
+  The ID of the VPC used for security group and internet gateway.
+  Required is `ec2_instance_create_associate_external_sg` is `true` or `ec2_instance_create_associate_igw` is `true`.
+
+* **ec2_instance_create_external_sg_id**: (Optional)
+  The ID or name of the security group to be associated with EC2 instance.
+
+* **ec2_instance_create_tags**: (Optional)
+  A dictionary of tags to assign to the EC2 instance.
+
+* **ec2_instance_create_wait_for_boot**: (Optional)
+  Whether to wait for the EC2 instance to be in the "running" state before continuing. Default is `true`.
+
+### Optional Networking Resources
+
+#### Elastic IP
+
+* **ec2_instance_create_associate_eip**: (Optional)
+  Whether to create an Elastic IP (EIP) and associate it with the EC2 instance. Default is `false`.
+  If set to `true` and provided VPC doesn't have an Internet Gateway (IGW) attached, please set `ec2_instance_create_associate_igw` to true to avoid failure due to VPC not having IGW attached.
+
+* **ec2_instance_create_eip_tags**: (Optional)
+  Tags to assign to the elastic IP.
+
+#### Internet Gateway
+
+* **ec2_instance_create_associate_igw**: (Optional)
+  Whether to create and associate an internet gateway with the EC2 instance. Default is `false`.
+  If set to `true`, an internet gateway will be created or associated with the instance.
+
+* **ec2_instance_create_igw_tags**: (Optional)
+  Tags to assign to the internet gateway.
+
+#### External Security Group
+
+* **ec2_instance_create_associate_external_sg**: (Optional)
```

Comment:
Does this replace or add to the security group defined in `ec2_instance_create_external_sg_id`? It might be helpful to clarify what the relationship is between these two security groups.

Reply:
updated readme.